src/cpp/session/modules/RStudioAPI.cpp

```diff
@@ -97,7 +97,7 @@ SEXP rs_showDialog(SEXP titleSEXP,
          return R_NilValue;
       }
 
-      if (question && !request.params[1].is_null()) {
+      if (dialogIcon == 4 && !request.params[1].is_null()) {
```

Comment:
Nit: consider mirroring enum on server so the meaning of "4" is clearer when reading this code